ipatests/test_integration/test_trust.py

```diff
@@ -992,3 +992,264 @@ def test_server_option_with_unreachable_ad(self):
+            ("cn", "ranges"),
+            ("cn", "etc"),
+            self.master.domain.basedn,
+        )
+        conn = self.master.ldap_connect()
+        entry = conn.get_entry(range_dn)  # pylint: disable=no-member
+        baseid = entry.single_value.get("ipaBaseID")
+        range = entry.single_value.get("ipaIDRangeSize")
+        cal_baseid = int(baseid) + int(range) + add_range
+        newbaseid = str(cal_baseid)
+        cmd = [
+            "ipa",
+            "idrange-add",
+            rangename,
+            "--base-id",
+            newbaseid,
+            "--range-size",
+            range,
+            "--rid-base",
+            newbaseid,
+            "--dom-name",
+            domainname,
+        ]
+
+        cmd.extend(("--type", trust_type))
+        cmd.extend(("--auto-private-groups", auto_priv_grp_val))
+
+        self.master.run_command(cmd)
+        self.master.run_command(["systemctl", "restart", "sssd"])
+
+    def mod_idrange_auto_private_group(
+        self, rangename, auto_priv_grp_val='false'
+    ):
+        """
+        Function to modify existing idrange
+        """
+        valid_msg = "Auto private groups: "
+        error_msg = "ipa: ERROR: no modifications to be performed"
+        rangename = self.ad_domain.upper() + "_id_range"
+        cmd = ["ipa", "idrange-mod", rangename]
+        cmd.extend(("--auto-private-groups", auto_priv_grp_val))
+        result = self.master.run_command(cmd, raiseonerr=False)
+        if result.returncode == 0:
+            assert valid_msg in result.stdout_text
+        else:
+            assert error_msg in result.stderr_text
+        self.master.run_command(["systemctl", "restart", "sssd"])
+
+    def test_add_idrange_auto_private_groups_false(self):
+        """
+        Test adds additional range with auto-private-groups value
+        as False and ensures that change is reflected in the output.
+        """
+        self.add_idrange_auto_private_group("testrange2")
+
+    def test_modify_auto_private_groups_value_default_trusted_range(self):
+        """
+        Modify existing range for default trusted AD domain range
+        with auto-private-groups set as true
+        """
+        domainusers = 'domain users'
+        nonposixuser = "nonposixuser@%s" % self.ad_domain
+        rangename = self.ad_domain.upper() + "_id_range"
+        self.mod_idrange_auto_private_group(rangename, "true")
+        result = self.master.run_command(["id", nonposixuser])
+        uid_reg = (
+            r"^uid=(\d+)\({0}\)".format(nonposixuser)
+        )
+        gid_reg = (
+            r"gid=(\d+)\({0}\)".format(nonposixuser)
+        )
+        grp_reg = (
+            r"groups=(\d+)\({0}\),(\d+)\({1}@{2}\)".format
+            (nonposixuser, domainusers, self.ad_domain)
+        )
+        assert re.search(uid_reg, result.stdout_text)
+        assert re.search(gid_reg, result.stdout_text)
+        assert re.search(grp_reg, result.stdout_text)
```

Comment:
I fail to understand the rationale for the 3 tests test_modify_auto_private_groups_value_default_trusted_range, test_modify_auto_private_groups_value_trusted_range_hybrid and test_idoverride_with_auto_private_group_false.
They have a different auto_private_group value but expect exactly the same thing (the output of `id username` should contain uid=xxx gid=xxx and groups=xxx).

You need to tailor each test and make clear that the expected value is different.